### PR TITLE
[mtouch] Remove most static state and put it on an instance.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Remove most static state, and put it on an instance (of Application) instead.

Soon we'll build multiple apps (app extensions) in the same process, and we
need to make sure we're not inadvertedly (and incorrectly) sharing state
between apps, so remove as much static state as possible to avoid any
problems.

### CODE REVIEW

File: tools/mtouch/Application.cs [hunk 1]
```diff
@@ -1812,7 +1828,7 @@ public static void Create (List<BuildTask> tasks, Abi abi, Target target, string
 
 		protected override void Build ()
 		{
-			if (Driver.IsUsingClang) {
+			if (Driver.GetIsUsingClang (App)) {
```
Comment (spouliot) [inline tools/mtouch/Application.cs]: `GetIs` sounds weird, refactored code ?

Comment (rolfbjarne) [inline tools/mtouch/Application.cs]: Heh, that's my refactoring autopilot 😄 I'll rename it to something better.

**Comment by rolfbjarne** [comment]
The mtouch test failures should be fixed with PR #1407.

**Comment by rolfbjarne** [comment]
Test failures are unrelated (build timeouts due to bug #[46097](http://bugzilla.xamarin.com/show_bug.cgi?id=46097))

**Pull request merged**